fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GrowthActivatedUserWorker.kt

```diff
@@ -0,0 +1,74 @@
+package org.mozilla.fenix.components.metrics
+
+import android.content.Context
+import androidx.work.ExistingWorkPolicy
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import mozilla.components.support.utils.ext.getPackageInfoCompat
+import org.mozilla.fenix.ext.metrics
+import org.mozilla.fenix.ext.settings
+import java.util.concurrent.TimeUnit
+
+/**
+ * Worker that will send the User Activated event at the end of the first week.
+ */
+class GrowthActivatedUserWorker(
+    context: Context,
+    workerParameters: WorkerParameters,
+) : Worker(context, workerParameters) {
+
+    override fun doWork(): Result {
+        val settings = applicationContext.settings()
+
+        if (!System.currentTimeMillis().isEndOfFirstWeek(applicationContext) ||
+            settings.growthUserActivatedSent
+        ) {
+            return Result.success()
+        }
+
+        applicationContext.metrics.track(Event.GrowthData.UserActivated(fromSearch = false))
+
+        return Result.success()
+    }
+
+    companion object {
+        private const val GROWTH_USER_ACTIVATED_WORK_NAME = "org.mozilla.fenix.growth.work"
+        private const val dayMillis: Long = 1000 * 60 * 60 * 24
+        private const val fullWeekMillis: Long = dayMillis * 7
+
+        /**
+         * Schedules the Activated User event if needed.
+         */
+        fun sendActivatedSignalIfNeeded(context: Context) {
+            val instanceWorkManager = WorkManager.getInstance(context)
+
+            if (context.settings().growthUserActivatedSent) {
+                return
+            }
+
+            val growthSignalWork = OneTimeWorkRequest.Builder(GrowthActivatedUserWorker::class.java)
+                .setInitialDelay(fullWeekMillis, TimeUnit.MILLISECONDS)
+                .build()
+
+            instanceWorkManager.beginUniqueWork(
+                GROWTH_USER_ACTIVATED_WORK_NAME,
+                ExistingWorkPolicy.KEEP,
+                growthSignalWork,
+            ).enqueue()
+        }
+
+        /**
+         * Returns [Boolean] value signaling if current time is after the first week after install.
+         */
+        private fun Long.isEndOfFirstWeek(context: Context): Boolean {
```

Comment:
nit, maybe:
```suggestion
        private fun Long.isAfterFirstWeekFromInstall(context: Context): Boolean {
```

Reply:
updated function name to be more descriptive